Code/OpenRealEstate.Core/Guard.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+
+namespace OpenRealEstate.Core
+{
+    public class Guard
+    {
+        public static void AgainstNulls(object obj)
+        {
+            if (obj == null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+        }
+
+        public static void AgainstNulls(string stringObj)

```

Comment:
Check if `WhiteSpace` or `Whitespace` (whatever the method if called on `string`)